--- a/drivers/mtd/nand/Kconfig
+++ b/drivers/mtd/nand/Kconfig
@@ -532,4 +532,16 @@ config MTD_NAND_FSMC
 	  Enables support for NAND Flash chips on the ST Microelectronics
 	  Flexible Static Memory Controller (FSMC)
 
+config MTD_NAND_P4A
+	tristate "P4A Nand Controller Driver"
+	depends on ARCH_P4A
+	help
+	  This is a nand controller driver for Innofidei P4A platform.
+
+config MTD_MAILBOX_NAND_P4A
+	tristate "Mailbox based Nand on P4A"
+	depends on ARCH_P4A && P4A_MAILBOX
+	help
+	  This is a driver for mailbox based nand on Innofidei P4A platform.
+
 endif # MTD_NAND
--- a/drivers/mtd/nand/Makefile
+++ b/drivers/mtd/nand/Makefile
@@ -48,5 +48,7 @@ obj-$(CONFIG_MTD_NAND_BCM_UMI)		+= bcm_umi_nand.o nand_bcm_umi.o
 obj-$(CONFIG_MTD_NAND_MPC5121_NFC)	+= mpc5121_nfc.o
 obj-$(CONFIG_MTD_NAND_RICOH)		+= r852.o
 obj-$(CONFIG_MTD_NAND_JZ4740)		+= jz4740_nand.o
+obj-$(CONFIG_MTD_NAND_P4A)			+= p4a_nand.o
+obj-$(CONFIG_MTD_MAILBOX_NAND_P4A)	+= p4a_mbnand.o
 
 nand-objs := nand_base.o nand_bbt.o
